File: Gombang/Weekly Contest 408/3232_Find_if_Digit_Game_Can_Be_Won.cs
public class Solution
{
	public bool CanAliceWin(int[] nums)
	{

		int singleDigitSum = 0;
		int doubleDigitSum = 0;

		for (int i = 0; i < nums.Length; i++)
		{
			if (nums[i] < 10)
			{
				singleDigitSum += nums[i];
			}
			else
			{
				doubleDigitSum += nums[i];
			}
		}

		return singleDigitSum != doubleDigitSum;
	}
}

File: Lucy/2024/Weekly Contest 408/3232_Find_if_Digit_Game_Can_Be_Won.js
/**
 * 3232. Find if Digit Game Can Be Won
 *
 * topic: Array, Math
 * difficulty: Easy
 * date: 24.07.29(MON)~
 */

/**
 * @param {number[]} nums
 * @return {boolean}
 */
var canAliceWin = function (nums) {
  const singleDigitSum = calcSigleDigitSum(nums);
  const doubleDigitSum = calcDoubleDigitSum(nums);

  if (singleDigitSum === doubleDigitSum) return false;
  if (singleDigitSum > doubleDigitSum || doubleDigitSum > singleDigitSum)
    return true;
};

const calcSigleDigitSum = (nums) =>
  nums.filter((n) => n < 10 && n >= 1).reduce((acc, curr) => acc + curr, 0);

const calcDoubleDigitSum = (nums) =>
  nums.filter((n) => n >= 10 && n <= 99).reduce((acc, curr) => acc + curr, 0);

console.log(canAliceWin([5, 5, 5, 25]));
